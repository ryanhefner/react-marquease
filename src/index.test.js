import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Marquee from './index'

describe('react-marquease', () => {
  beforeAll(() => {
    window.innerWidth = 1024
  })

  it('renders', async () => {
    const { asFragment } = render(<Marquee>Test</Marquee>)

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders 2 clones', async () => {
    const { getAllByText } = render(<Marquee>Test</Marquee>)

    await getAllByText('Test')
    expect(getAllByText('Test')).toHaveLength(3)
  })
})
