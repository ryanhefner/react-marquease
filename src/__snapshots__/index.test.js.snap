// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-marquease renders 1`] = `
<DocumentFragment>
  <div
    style="position: relative; height: 0px;"
  >
    <div
      style="position: absolute; top: -9999px; left: -9999px;"
    >
      Test
    </div>
    <div
      data-qa="scroller"
      style="transform: translateX(0px);"
    >
      <div
        data-marquee-child="0"
        style="position: absolute; top: 0px; transform: translateX(0px);"
      >
        Test
      </div>
      <div
        data-marquee-child="1"
        style="position: absolute; top: 0px; transform: translateX(0px);"
      >
        Test
      </div>
    </div>
  </div>
</DocumentFragment>
`;
